.github/workflows/main.yml

```diff
@@ -35,18 +36,64 @@ jobs:
 
       - name: Save docker image
         run: |
-          docker save research-template | gzip > /tmp/research-template.tar.gz
+          docker save research-template | pigz --fast > /tmp/research-template.tar.gz
 
       - name: Upload docker image
         uses: actions/upload-artifact@v4
         with:
             name: research-template-image
             path: /tmp/research-template.tar.gz
+            # Disable compression; the file is already compressed
+            compression-level: 0
 
-  publish:
+  test-devcontainer:
     needs: [build-and-test]
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Download Docker image
+      uses: actions/download-artifact@v4
+      with:
+        name: research-template-image
+        path: /tmp/image
+
+    - name: Import Docker image
+      run: docker load --input /tmp/image/research-template.tar.gz
+
+    - name: Tag Docker image for use with dev container
+      run: docker tag "$IMAGE_NAME" "$PUBLIC_IMAGE_NAME:$IMAGE_VERSION"
```

Comment:
Why are we tagging it with the version we're building? Can't we tag it with a `dev` version?

Reply:
We could, if that's really helpful to avoid any confusion.

The downside with that is we then need an extra step of ensuring that the `v0` or similar image version in `devcontainer.json` is also substituted with `dev`.

Reply:
I think I've warmed to this as an idea; see [my other comment for the reasoning](https://github.com/opensafely-core/research-template-docker/pull/13#discussion_r1610370250).

Reply:
Although I should think about how that interacts with an automated scheduled test.

Reply:
Have done this.